### PR TITLE
[ZEPPELIN-5990] Disable sensitive configuration for JDBC url

### PR DESCRIPTION
**Pull request by jongyoul**
### What is this PR for?
Disabling some sensitive configuration for JDBC URL when running queries.

### What type of PR is it?
Feature

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-5990

### How should this be tested?
* Test should be passed
* Sensitive configuration shouldn't be accepted

### Questions:
* Does the license files need to update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by jongyoul** [comment]
ping @zjffdu @Reamer @huage1994

**Review by zjffdu — APPROVED** [review]

**Comment by pjfanning** [comment]
This is a fix for CVE-2024-31864
I don't believe a CVE should have been issued for this and it has opened up a long list of requests for similar CVEs.
I see no problem with hardening the Zeppelin code, I just disagree with assigning CVEs for things like this.
If you read https://zeppelin.apache.org/security.html
* Zeppelin is all about letting you run code
* You are expected to run Zeppelin on a secure network or to enable Shiro auth if you need to open up access
* In short, Zeppelin is not expected to be run in an adversarial environment
* MySQL jars are not shipped with Zeppelin - we should just say that users who add JDBC driver jars to Zeppelin are doing so at their own risk

fyi @jongyoul @zjffdu @Reamer @raboof
